check_limits.py

```diff
@@ -1,5 +1,29 @@
+ranges ={
+  'temp':{'min':0,'max':45},#Temperature
+  'soc':{'min':20,'max':80},#state of Charge
+  'charge_rate':{'min':0,'max':0.8},#state of Charge
+  }
+test_report=[]
+def collect_abnormals(abnormals,attribute_name,attribute_value,attribute_range):
+  if attribute_value < attribute_range['min'] or attribute_value > attribute_range['max']:
```

Comment:
The report doesn't state if it is high or low. Try adding that. You have isolated the check, so it should be easy.